File: examples-sdk/go/605-db-vilorm-crud/main.go
// 605 — Blog Platform (VilORM CRUD)
// Equivalent to: examples/605-db-vilorm-crud (Rust)
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	server := vil.NewServer("vilorm-showcase", 8080)

	blog := vil.NewService("blog")
	blog.Endpoint("POST", "/authors", "create_author")
	blog.Endpoint("GET", "/authors", "list_authors")
	blog.Endpoint("GET", "/posts", "list_posts")
	blog.Endpoint("POST", "/posts", "create_post")
	blog.Endpoint("GET", "/posts/:id", "get_post")
	blog.Endpoint("PUT", "/posts/:id", "update_post")
	blog.Endpoint("DELETE", "/posts/:id", "delete_post")
	blog.Endpoint("POST", "/tags", "create_tag")
	blog.Endpoint("GET", "/stats", "blog_stats")
	server.Service(blog)

	server.Compile()
}

File: examples-sdk/zig/605-db-vilorm-crud/main.zig
// 605-db-vilorm-crud — Zig SDK equivalent
// Compile: vil compile --from zig --input 605-db-vilorm-crud/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("vilorm-showcase", 8080);
    var blog = vil.Service.init("blog");
    blog.endpoint("POST", "/authors", "create_author");
    blog.endpoint("GET", "/authors", "list_authors");
    blog.endpoint("GET", "/posts", "list_posts");
    blog.endpoint("POST", "/posts", "create_post");
    blog.endpoint("GET", "/posts/:id", "get_post");
    blog.endpoint("PUT", "/posts/:id", "update_post");
    blog.endpoint("DELETE", "/posts/:id", "delete_post");
    blog.endpoint("POST", "/tags", "create_tag");
    blog.endpoint("GET", "/stats", "blog_stats");
    server.service(&blog);
    server.compile();
}
